mantidproject/mantid
# Eng Diff Fitting Tab - Loading Files

**Description of work.**
The initial framework of the Engineering Diffraction 2 GUI's Fitting tab. Allows for the loading of one to many files into mantid. 
The GUI keeps track of the files it loaded using the `AnalysisDataServiceObserver` class. 
The "Add to Plot" check box currently does nothing. 
The GUI does not currently check that the files being loaded are suitable for the GUI. 
**To test:**
1. Add this line to "mantid.user.properties": `mantidqt.python_interfaces=Direct/DGS_Reduction.py Direct/DGSPlanner.py Direct/PyChop.py Direct/MSlice.py SANS/ORNL_SANS.py Utility/TofConverter.py Reflectometry/ISIS_Reflectometry_Old.py Diffraction/Engineering_Diffraction_2.py Diffraction/Powder_Diffraction_Reduction.py Utility/FilterEvents.py Diffraction/HFIR_4Circle_Reduction.py Utility/QECoverage.py SANS/ISIS_SANS.py Muon/Frequency_Domain_Analysis.py Muon/Elemental_Analysis.py Muon/Frequency_Domain_Analysis_Old.py Muon/Muon_Analysis.py` to allow access to the Engineering Diffraction 2 GUI.
1. Open the Engineering Diffraction 2 GUI (Interfaces -> Diffraction -> Engineering Diffraction 2)
1. Perform a calibration (unless you have one already, then load that one) using V:307521 S: 305738
2. Perform a focus for each bank so you have a range of files. S: 305761
3. Open the fitting tab 
4. Load a range of file names and types.
5. Check that all selected files were loaded.
6. Make sure the Messages Log is in debug mode. 
7. Rename, load more, and delete some workspaces and check that the logged list matches the expected list of workspaces BEFORE the operation. 

Fixes #27777 

*This does not require release notes* because **the GUI is not user facing yet**.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
